gobblin-yarn/src/main/java/gobblin/yarn/YarnService.java

```diff
@@ -446,13 +485,17 @@ private void handleContainerCompletion(ContainerStatus containerStatus) {
     @Override
     public void onContainersCompleted(List<ContainerStatus> statuses) {
       for (ContainerStatus containerStatus : statuses) {
-        handleContainerCompletion(containerStatus);
+        int retryCount = handleContainerCompletion(containerStatus);
+        eventSubmitter.submit(GobblinYarnEventConstants.EventNames.CONTAINER_COMPLETION,
```

Comment:
You can move this call to inside `handleContainerCompletion`, then you don't need to change the signature of `handleContainerCompletion`.